src/lib.rs

```diff
@@ -122,76 +125,64 @@ impl UnixListener {
     /// This function will panic if it is called outside the context of a
     /// future's task. It's recommended to only call this from the
     /// implementation of a `Future::poll`, if necessary.
-    pub fn accept(&mut self) -> io::Result<(UnixStream, SocketAddr)> {
-        loop {
-            if let Some(mut pending) = self.pending_accept.take() {
-                match pending.poll().expect("shouldn't be canceled") {
-                    Async::NotReady => {
-                        self.pending_accept = Some(pending);
-                        return Err(would_block())
-                    },
-                    Async::Ready(r) => return r,
-                }
-            }
+    pub fn poll_accept(&self) -> Poll<(UnixStream, SocketAddr), io::Error> {
+        let (io, addr) = try_ready!(self.poll_accept_std());
 
-            if let Async::NotReady = self.io.poll_read() {
-                return Err(io::Error::new(io::ErrorKind::WouldBlock, "not ready"))
-            }
+        let io = PollEvented::new(io);
+        Ok((UnixStream { io: io }, addr).into())
+    }
+
+    /// Attempt to accept a connection and create a new connected `UnixStream`
+    /// if successful.
+    ///
+    /// This function is the same as `poll_accept` above except that it returns a
+    /// `mio_uds::UnixStream` instead of a `tokio_udp::UnixStream`. This in turn
+    /// can then allow for the stream to be associated with a different reactor
+    /// than the one this `UnixListener` is associated with.
+    ///
+    /// This function will attempt an accept operation, but will not block
+    /// waiting for it to complete. If the operation would block then a "would
+    /// block" error is returned. Additionally, if this method would block, it
+    /// registers the current task to receive a notification when it would
+    /// otherwise not block.
+    ///
+    /// Note that typically for simple usage it's easier to treat incoming
+    /// connections as a `Stream` of `UnixStream`s with the `incoming` method
+    /// below.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if it is called outside the context of a
+    /// future's task. It's recommended to only call this from the
+    /// implementation of a `Future::poll`, if necessary.
+    pub fn poll_accept_std(&self) -> Poll<(mio_uds::UnixStream, SocketAddr), io::Error> {
```

Comment:
It is a bit weird to have it named `_std` when it actually returns a `mio` type but I suppose it is going to be inconsistent one way or another.

Reply:
Oh shoot, I missed that.

Is there any way that function can return https://doc.rust-lang.org/std/os/unix/net/struct.UnixStream.html?

Reply:
Ideally, this library would not expose any types from `mio_uds`.

Reply:
Filed https://github.com/alexcrichton/mio-uds/pull/11 to add `accept_std` to mio-uds

Reply:
Fixed